### PR TITLE
Fix mixed x86/amd64 Linux platforms

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: mesonbuild/dependencies.py [hunk 1]
```diff
@@ -721,8 +724,13 @@ def get_link_args(self):
...
+            libname = 'boost_' + module
+            # The compiler's library detector is the most reliable so use that first.
+            default_detect = self.cpp_compiler.find_library(libname, self.environment, [])
+            if default_detect is not None:
+                args += default_detect
+            elif module in self.lib_modules or module in self.lib_modules_mt:
+                linkcmd = '-l' + libname
                 args.append(linkcmd)
                 # FIXME a hack, but Boost's testing framework has a lot of
                 # different options and it's hard to determine what to do
```
Comment (nirbheek) [inline mesonbuild/dependencies.py]: Shouldn't the `unit_testing_framework` section below always be checked?

**Comment by jpakkane** [comment]
Updated.

File: mesonbuild/dependencies.py [hunk 2]
```diff
@@ -721,8 +724,16 @@ def get_link_args(self):
             args.append('-L' + os.path.join(self.boost_root, 'lib'))
         for module in self.requested_modules:
             module = BoostDependency.name2lib.get(module, module)
-            if module in self.lib_modules or module in self.lib_modules_mt:
-                linkcmd = '-lboost_' + module
+            libname = 'boost_' + module
+            # The compiler's library detector is the most reliable so use that first.
+            default_detect = self.cpp_compiler.find_library(libname, self.environment, [])
+            if default_detect is not None:
+                emon_args = self.cpp_compiler.find_library('boost_test_exec_monitor')
```
Comment (nirbheek) [inline mesonbuild/dependencies.py]: This will add it unconditionally and repeatedly for each module; is this what we want?

**Comment by jpakkane** [comment]
Crap. Good catch, fixed.

**Comment by jpakkane** [comment]
In theory yes, but the recommended way to use gtest and gmock is to not use precompiled libraries but instead compile yourself from source. (at least, this was the recommendation at some point, it might have changed, or rechanged, we'll fix it if anyone actually files a bug or something breaks)

**Pull request merged**